### PR TITLE
TSV Invalid Attribute Validation

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
- Validate the case for an attribute with no value
- Validate the case for an attribute detail without parent attribute
- Add corresponding unit tests

### CODE REVIEW

File: serialization/src/test/kotlin/ac/uk/ebi/biostd/tsv/TsvDeserializerTest.kt [hunk 1]
```diff
@@ -218,6 +227,28 @@ class TsvDeserializerTest {
...
+        assertInvalidElementException(assertThrows {
+            deserializer.deserialize(submissionWithInvalidNameAttributeDetail().toString()) }, MISPLACED_ATTR_NAME)
+
+    @Test
+    fun deserializeInvalidValueAttributeDetail() =
+        assertInvalidElementException(assertThrows {
+            deserializer.deserialize(submissionWithInvalidValueAttributeDetail().toString()) }, MISPLACED_ATTR_VAL)
+
+    private fun assertInvalidElementException(exception: SerializationException, expectedMessage: String) {
+        val cause = exception.errors.values().first().cause
```
Comment (Juan-EBI) [inline serialization/src/test/kotlin/ac/uk/ebi/biostd/tsv/TsvDeserializerTest.kt]: this make result in null pointer exception or invalid index exception so better add before

```
asssertThat(exception.errors).hasSize(1)

```

to get proper error message in test

File: serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt [hunk 2]
```diff
@@ -12,21 +15,29 @@ internal inline fun validate(value: Boolean, lazyMessage: () -> String) {
...
-            line.isValueDetail() -> attributes.last().valueAttrs.add(AttributeDetail(line.name(), line.value))
+            line.isNameDetail() -> addNameAttributeDetail(attributes, line)
+            line.isValueDetail() -> addValueAttributeDetail(attributes, line)
             else -> attributes.add(Attribute(line.name(), line.value, line.isReference()))
         }
     }
     return attributes
 }
 
+internal fun addNameAttributeDetail(attributes: MutableList<Attribute>, line: TsvChunkLine) {
```
Comment (Juan-EBI) [inline serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt]: why not private

File: serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt [hunk 3]
```diff
@@ -12,21 +15,29 @@ internal inline fun validate(value: Boolean, lazyMessage: () -> String) {
...
+    attributes.ifEmpty { throw InvalidElementException(MISPLACED_ATTR_NAME) }
+    attributes.last().nameAttrs.add(AttributeDetail(line.name(), line.value))
+}
+
+internal fun addValueAttributeDetail(attributes: MutableList<Attribute>, line: TsvChunkLine) {
```
Comment (Juan-EBI) [inline serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt]: why not private

File: serialization/src/test/kotlin/ac/uk/ebi/biostd/tsv/TsvDeserializerTest.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply serialization/src/test/kotlin/ac/uk/ebi/biostd/tsv/TsvDeserializerTest.kt]: Done

File: serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt (hunk 2 above)
Comment (jhoanmanuelms) [reply serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt]: Fixed

File: serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt (hunk 3 above)
Comment (jhoanmanuelms) [reply serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/common/Functions.kt]: Fixed

**Pull request merged**